x-pack/plugins/security_solution/public/common/components/ml_popover/api.ts

```diff
@@ -50,12 +52,17 @@ export const checkRecognizer = async ({
  *
  * @throws An error if response is not OK
  */
-export const getModules = async ({ moduleId = '', signal }: GetModulesProps): Promise<Module[]> =>
+export const getModules = async ({
+  moduleId = '',
+  signal,
+  moduleFilter,
+}: GetModulesProps): Promise<Module[]> =>
   KibanaServices.get().http.fetch<Module[]>(`/internal/ml/modules/get_module/${moduleId}`, {
     method: 'GET',
     version: '1',
     asSystemRequest: true,
     signal,
+    query: { filter: moduleFilter },
```

Comment:
nit: As `getModules` defined inside Security Solution there is a low chance `search` and `observability` ML modules will be fetched by using this function. I'd just inline `{ filter: 'security'}` here and it will make the diff even smaller.

Reply:
Updated in https://github.com/elastic/kibana/pull/168721/commits/322b99b191c14a5db06713c8f70226d6a7a9e1a9